compiler bug when panic is used in both else clauses of two match statements added together, even if the else statement is impossible/not run

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: macos, macOS, 12.0.1, 21A559
Processor: 10 cpus, 64bit, little endian, Apple M1 Max
CC version: Apple clang version 13.0.0 (clang-1300.0.29.3)

getwd: /Users/thomas/v-coding/
vmodules: /Users/thomas/.vmodules
vroot: /Users/thomas/Unix/v-weekly
vexe: /Users/thomas/Unix/v-weekly/v
vexe mtime: 2021-11-19 00:27:43
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 24ba660

Git version: git version 2.34.0
Git vroot status: weekly.2021.46-38-g24ba6603
.git/config present: true
thirdparty/tcc status: thirdparty-macos-amd64 689c8a02

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg bug.v`
```v
module main

fn hex_to_bytes (s string) ?[]byte {
	mut ret := []byte{cap: s.len}
	for read := 0; read < s.len; read += 2 {
		high := s[read]
		low := s[read+1]

		ret << match high {
			48...57 { (high - 48) << 4 }
			65...70 { (high - 65) << 4 }
			97...102 { (high - 97) << 4 }
			else { panic("impossible") }
		} + match low {
			48...57 { low - 48 }
			65...70 { low - 65 }
			97...102 { low - 97 }
			else { panic("impossible") }
		}
	}
	return ret
}

fn main () {
	println(hex_to_bytes('FFFF') or { 'error'.bytes() })
}

```

**What did you expect to see?**

the program returning the []bytes with 0xff 0xff


**What did you see instead?**
```
==================
/tmp/v_501/bug.14326221822549201433.tmp.c:10660:50: error: use of undeclared identifier '_t3'; did you mean '_t2'?
                }array_push((array*)&ret, _MOV((byte[]){ _t2 + _t3 }));
                                                               ^~~
                                                               _t2
/tmp/v_501/bug.14326221822549201433.tmp.c:10632:8: note: '_t2' declared here
                byte _t2 = 0;
                     ^
1 error generated.
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -4444,6 +4444,7 @@ fn (mut g Gen) match_expr(node ast.MatchExpr) {
 	} else {
 		g.match_expr_classic(node, is_expr, cond_var, tmp_var)
 	}
+	g.set_current_pos_as_last_stmt_pos()
 	g.write(cur_line)
 	if need_tmp_var {
 		g.write('$tmp_var')

--- a/vlib/v/tests/multiple_matchs_in_one_expr_test.v
+++ b/vlib/v/tests/multiple_matchs_in_one_expr_test.v
@@ -1,0 +1,26 @@
+fn hex_to_bytes(s string) ?[]byte {
+	mut ret := []byte{cap: s.len}
+	for read := 0; read < s.len; read += 2 {
+		high := s[read]
+		low := s[read + 1]
+
+		ret << match high {
+			48...57 { (high - 48) << 4 }
+			65...70 { (high - 65) << 4 }
+			97...102 { (high - 97) << 4 }
+			else { panic('impossible') }
+		} + match low {
+			48...57 { low - 48 }
+			65...70 { low - 65 }
+			97...102 { low - 97 }
+			else { panic('impossible') }
+		}
+	}
+	return ret
+}
+
+fn test_multiple_matchs_in_one_expr() {
+	ret := hex_to_bytes('FFFF') or { 'error'.bytes() }
+	println(ret)
+	assert '$ret' == '[U, U]'
+}

